infra/license/license-checker.ts

```diff
@@ -0,0 +1,179 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {readFileSync, writeFileSync} from 'fs';
+import {EOL} from 'os';
+
+import * as licenseJudgment from './license-judgment.json';
+import * as packageJudgment from './package-judgment.json';
+
+
+/**
+ * Argument List
+ *  - Input path for list of used licenses
+ *  - Name of OS.
+ *    Currently, latest version of Winodws, Ubuntu, MacOS are used.
+ */
+var args = process.argv.slice(2)
+const inputPathOfLicenseList = args[0];
+const verificationTag = args[1];
+
+/**
+ * - WarningCount : The total number of warnings
+ *
+ * - WarnnedLicenseUsed
+ *   - Licenses which are classified as WARN in license-judgment.json
+ * - NeverChecked
+ *   - Licenses which are not found in license-judgment.json
+ * - UnknownLicense
+ *   - Licenses which package are not found.
+ */
+var warningList = {
+  'WarningCount': 0,
+
+  'WarnedLicenseUsed': [] as string[],
+  'NeverChecked': [] as string[],
+  'UnknownLicense': [] as string[]
+};
+
+/**
+ * - DeniedCount : The total number of denials
+ *
+ * - ONEForbidden
+ *   - Licenses which are classified as not permitted at ONE in package-judgment.json
+ * - DeniedLicenseUsed
+ *   - Licenses which are denied by license-judgment.json
+ */
+var deniedList = {
+  'DeniedCount': 0,
+
+  'ONEForbidden': [] as string[],
+  'DeniedLicenseUsed': [] as string[]
+};
+
+/**
+ * License Verification start
+ *
+ * NOTE : "inputPathOfLicenseList" includes following JSON content,
+ *        which is result of 'license-checker'
+ * {
+ *   "one-vscode@0.0.1": {
+ *     "licenses": "UNKNOWN",
+ *     "repository": "https://github.com/Samsung/ONE-vscode",
+ *     "dependencyPath": "/path/to/ONE-vscode/",
+ *     "path": "/path/to/ONE-vscode/",
+ *     "licenseFile": "ONE-vscode/LICENSE"
+ *   },
+ *   "one-vscode@0.0.2": {
+ *   ...
+ * }
+ */
+const usedLicenseList = JSON.parse(readFileSync(inputPathOfLicenseList, 'utf-8'));
+for (const pkg in usedLicenseList) {
+  var pkgInfo = usedLicenseList[pkg];
+  if (packageJudgment.hasOwnProperty(pkg)) {
+    const pkgKey = pkg as keyof typeof packageJudgment;
+
+    if (packageJudgment[pkgKey].ONE_permitted === 'no') {
+      deniedList.DeniedCount++;
+      deniedList.ONEForbidden.push(pkg + EOL);
+    } else if (packageJudgment[pkgKey].ONE_permitted === 'yes') {
+      // Verification PASS, do nothing
+    } else if (packageJudgment[pkgKey].ONE_permitted === 'conditional') {
```

Comment:
Note that this one needs to be manually checked during release process.
example is `argparse@2.0.1` and condition is described at `ONE_caution` in `package-judgment.json`.